x-pack/legacy/plugins/apm/public/components/shared/LocalUIFilters/Filter/index.tsx

```diff
@@ -0,0 +1,168 @@
+  background: ${theme.euiColorLightShade};
+  padding: 0 ${theme.paddingSizes.xs};
+`;
+
+const ApplyButton = styled(EuiButton)`
+  align-self: flex-end;
+`;
+
+interface Props {
+  name: string;
+  title: string;
+  options: Array<{
+    name: string;
+    count: string;
+  }>;
+  onChange: (value: string[]) => void;
+  value: string[];
+  showCount: boolean;
+}
+
+type Option = EuiSelectable['props']['options'][0];
+
+const Filter = ({
+  name,
+  title,
+  options,
+  onChange,
+  value,
+  showCount
+}: Props) => {
+  const [showPopover, setShowPopover] = useState(false);
+
+  const toggleShowPopover = () => setShowPopover(show => !show);
+
+  const button = (
+    <FilterTitleButton onClick={toggleShowPopover}>{title}</FilterTitleButton>
+  );
+
+  const items: Option[] = useMemo(
+    () =>
+      options.map(option => ({
+        label: option.name,
+        append: showCount ? (
+          <Counter>
+            <EuiText size="xs">{option.count}</EuiText>
+          </Counter>
+        ) : null,
+        checked: value.includes(option.name) ? 'on' : undefined
+      })),
+    [value, options, showCount]
+  );
+
+  const [visibleOptions, setVisibleOptions] = useState(items);
+
+  useEffect(() => {
+    setVisibleOptions(items);
+  }, [items]);
+
+  return (
+    <>
+      <Popover
+        id={`local-filter-${name}`}
+        closePopover={toggleShowPopover}
+        button={button}
+        isOpen={showPopover}
+        panelPaddingSize="s"
+        anchorPosition="downLeft"
+      >
+        <EuiSelectable
+          onChange={selectedOptions => {
+            setVisibleOptions(selectedOptions);
+          }}
+          options={visibleOptions}
+          searchable={true}
+        >
+          {(list, search) => (
+            <SelectContainer>
+              <EuiFlexGroup direction="column" gutterSize="none">
+                <EuiFlexItem grow={true}>
+                  <EuiTitle size="xxxs" textTransform="uppercase">
+                    <h4>
+                      {i18n.translate('xpack.apm.applyFilter', {
+                        defaultMessage: 'Apply {title} filter',
+                        values: { title }
+                      })}
+                    </h4>
+                  </EuiTitle>
+                  <EuiHorizontalRule margin="none" />
+                  <EuiSpacer size="s" />
+                  {search}
+                  <EuiSpacer size="s" />
+                  {list}
+                  <EuiSpacer size="s" />
+                </EuiFlexItem>
+                <EuiFlexItem grow={false}>
+                  <ApplyButton
+                    color="primary"
+                    fill={true}
+                    onClick={() => {
+                      const newValue = visibleOptions
+                        .filter(option => option.checked === 'on')
+                        .map(option => option.label);
+
+                      setShowPopover(false);
+                      onChange(newValue);
```

Comment:
Perhaps this should be called `onApply`. I was initally a little confused about the `onChange` handler on `EuiSelectable` and this.

Reply:
I called this onChange because the concept of applying is specific to this component. For instance, with a dropdown, there's really no apply, just a change. Happy to change this if you still feel it makes things more clear though.

Reply:
I don't feel strongly. Was only if you had ideas to make it a little more obvious to the reader. I'm fine with it as-is.